packages/clerk-js/src/ui/customizables/makeResponsive.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import { isDataUri, isValidUrl } from '../../utils';
+
+type Responsive<T = Record<never, never>> = T & {
+  size?: number;
+  xDescriptors?: number[];
+};
+
+type ResponsivePrimitive<T> = React.FunctionComponent<Responsive<T>>;
+
+export const makeResponsive = <P extends React.JSX.IntrinsicElements['img']>(
+  Component: React.FunctionComponent<P>,
+): ResponsivePrimitive<P> => {
+  const responsiveComponent = React.forwardRef((props: Responsive<any>, ref) => {
+    const { src, size = 80, xDescriptors = [1, 2], ...restProps } = props;
+    const shouldOptimize = isClerkImage(src);
+
+    return (
+      <Component
+        srcSet={shouldOptimize ? generateSrcSet({ src, width: size, xDescriptors }) : undefined}
+        src={shouldOptimize ? generateSrc({ src, width: size * 2 }) : src}
+        {...restProps}
+        ref={ref}
+      />
+    );
+  });
+
+  const displayName = Component.displayName || Component.name || 'Component';
+  responsiveComponent.displayName = `Responsive${displayName}`.replace('_', '');
+  return responsiveComponent as ResponsivePrimitive<P>;
+};
+
+const CLERK_IMAGE_URL_BASES = [
+  'https://img.clerk.com/',
+  'https://img.clerk.dev/',
+  'https://img.clerkstage.dev/',
+  'https://img.lclclerk.com/',
+];
```

Comment:
🙃 i would keep it in `shared` (and also keep the `isClerkImage` and the below Framework agnostic utilities).

Reply:
This makes sense to me, however, I think these are specific enough to have them colocated with the HOC